src/utilities/ivao.service.ts

```diff
@@ -2,37 +2,127 @@ import { HttpService, Injectable, Logger } from '@nestjs/common';
+        pilots: Client[];
+    };
+    connections: {
+        atc: number;
+        observer: number;
+        pilot: number;
+        supervisor: number;
+        total: number;
+        worldTour: number;
+    };
+}
+
+interface AtisMergedWhazzup extends Whazzup {
+    clients: {
+        atcs: (Client & { atis?: Atis })[];
+        followMe: unknown[];
+        observers: Client[];
+        pilots: Client[];
+    }
+}
+
+type WhazzupAtis = Atis[];
 
 @Injectable()
 export class IvaoService {
-  private readonly logger = new Logger(IvaoService.name);
-
-  constructor(
-    private http: HttpService,
-    private readonly cache: CacheService,
-  ) {}
-
-  public async fetchIvaoData(): Promise<string[]> {
-      const cacheHit = await this.cache.get<string[]>('/ivao/data');
-
-      if (cacheHit) {
-          this.logger.debug('Returning from cache');
-          return cacheHit;
-      }
-      const data = await this.http.get<Buffer>('https://api.ivao.aero/getdata/whazzup/whazzup.txt', { responseType: 'arraybuffer' })
-          .pipe(
-              tap((response) => this.logger.debug(`Response status ${response.status} for IVAO  request`)),
-              tap((response) => this.logger.debug(`Response contains ${response.data.length} entries`)),
-              map((response) => iconv
-                  .decode(response.data, 'ISO-8859-1')
-                  .split(/\r?\n/)),
-          ).toPromise();
-
-      this.cache.set('/ivao/data', data, 120).then();
-      return data;
-  }
+    private readonly logger = new Logger(IvaoService.name);
+
+    constructor(
+        private http: HttpService,
+        private readonly cache: CacheService,
+    ) {}
+
+    public async fetchIvaoData(): Promise<AtisMergedWhazzup> {
+        const cacheHit = await this.cache.get<AtisMergedWhazzup>('/ivao/data');
+
+        if (cacheHit) {
+            this.logger.debug('Returning from cache');
+            return cacheHit;
+        }
+
+        const whazzupData: Whazzup = await this.http.get('https://api.ivao.aero/v2/tracker/whazzup')
+            .pipe(
+                tap((response) => this.logger.debug(`Response status ${response.status} for IVAO  request`)),
+                tap((response) => this.logger.debug(`Response contains ${response.data.length} entries`)),
+                map((response) => response.data),
+            ).toPromise();
+
+        const whazzupAtisData: WhazzupAtis = await this.http.get('https://api.ivao.aero/v2/tracker/whazzup/atis')
+            .pipe(
+                tap((response) => this.logger.debug(`Response status ${response.status} for IVAO  request`)),
+                tap((response) => this.logger.debug(`Response contains ${response.data.length} entries`)),
+                map((response) => response.data),
+            ).toPromise();
+
+        for (const atis of whazzupAtisData) {
+            const atcWithSameCallsign = (whazzupData as AtisMergedWhazzup).clients.atcs.find((atcs) => atcs.callsign === atis.callsign);
```

Comment:
**question:** do we need this cast here ?

Reply:
Yes, the assertion is necessary because the type Whazzup does not have the property atis.